Testing/AutomatedA11yTesting.md

```diff
@@ -0,0 +1,79 @@
+
+For most pages we didn’t encounter any issues with performing our a11y tests. We were also able to easily filter out an a11y issue that was identified in the menu for the Web Experience Toolkit.
+
+We did however run into issues when attempting to automate the ROEWeb Assistant functionality. For the most part it worked fine but it’s asynchronous nature and the fact that the developer who wrote it (Calvin Rodo) did not design it with an automated tool in mind, meant we ran into several issues with this page.
+
+This really highlights the need for developers to be involved in writing the test automation for their sites, when developers aren’t involved they have no real incentive to write testable applications. When they are are responsible for the automated tests they feel the pain in attempting to automate their applications and will thus write more testable code and will invest more effort into maintaining and fixing the tests.
+
+## Recommendation from the Developer Experience Team
+
+We recommend that we attempt to automate the a11y testing of an application that is currently in development at ESDC to show the value of this type of automation.
+
+The ideal application would be one that is currently doing automated builds, and has developers that are comfortable writing code in JavaScript.
+
+## Requirements for Next Steps
+
+* NodeJS installed on Build Machines
+* Chromium installed on Build Machines
+  * It’s critical to the work that the department is doing that we have access to either Chromium or Chrome, most modern in browser development tools are being built for Chrome first. Without access to this tool we are crippling ourselves and making it harder to do modern web development.
+
+## Expected Outcome
+
+According to aXe their tool will only cover 20 to 50% of all accessibility issues. Automation of a11y testing through aXe coupled with an html5 validation library we should be able to free up the teams at ESDC assessing our sites for compliance with a11y requirements to perform more valuable non-automated work.
+
+Automated testing will also provide feedback on a11y issues earlier to developers so which provides instant learning opportunities and should over time increase the institutional knowledge of writing a11y web sites.
+
+CDS-SNC has some tools they are working on that are similar to puppeteer and axe-core, where they scan the front page of all Government of Canada Websites, something in the thousands.
+Here are some of the related projects from their GitHub:
+
+## cds-snc/domain-scan
+
+A lightweight pipeline, locally or in Lambda, for scanning things like HTTPS, third party service use, and web accessibility.
+<https://github.com/cds-snc/domain-scan>
+
+## cds-snc/lighthouse-scanner
+
+<https://github.com/cds-snc/lighthouse-scanner>
+
+## cds-snc/loghthouse-scan-monitor
+
+<https://github.com/cds-snc/lighthouse-scan-monitor>
+
+## cds/snc/lighthouse-scanner-ui
+
+<https://github.com/cds-snc/lighthouse-scanner-ui>
```

Comment:
As all of this was part of a comment, can we add a header (something like "Additional Tools") and just make the tools a list of links?
What do you think?

Reply:
Done.

Reply:
Line 61 + 62 are part of the "Additional Tools" so should also be moved under that header.